src/lib/flow/sol-flow-resolver-conffile.c

```diff
@@ -179,7 +179,10 @@ _resolver_conffile_get_module(const char *type)
     }
 
     entry = sol_vector_append(&resolver_conffile_dlopens);
-    SOL_NULL_CHECK(entry, NULL);
+    if (!entry) {
```

Comment:
ok, but if you want to keep the warning from `SOL_NULL_CHECK()` you can use the variant `SOL_NULL_CHECK_GOTO()` and put a label to do this cleanup and return NULL